src/ucp/wireup/select.c

```diff
@@ -2509,10 +2508,12 @@ ucp_wireup_select_lanes(ucp_ep_h ep, unsigned ep_init_flags,
 {
     ucp_worker_h worker                = ep->worker;
     ucp_tl_bitmap_t scalable_tl_bitmap = worker->scalable_tl_bitmap;
+    /* TO DO: remove initializtion after all ucp_wireup_add_X_lanes functions
+       will support specifying a reason */
+    char wireup_info[256]              = {0};
```

Comment:
do we really need to initialize, given that uct_iface_is_reachable_v2() already initializes it to NULL?

Reply:
Currently we do, because `ucp_wireup_add_am_lane` is the only one (out of all `ucp_wireup_add_*_lane` functions) that supports filling info_str. All the rest don't receive info_str and don't initialize it (until I'll add the other transports). So if we don't initialize it and one of them fails before `ucp_wireup_add_am_lane` is called, `ucp_wireup_search_lanes` returns a string filled with trash.

Reply:
ok
do you plan to add support for other ucp_wireup_add_*_lane in next pr?

Reply:
@yosefe yes.